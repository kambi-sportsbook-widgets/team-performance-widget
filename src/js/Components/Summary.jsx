import React from 'react';
import EventResultIndicator from './EventResultIndicator';
import ParticipantSummary from './ParticipantSummary';
import VersusIcon from './VersusIcon';

const Summary = ({ participants }) => {
   return (
      <div className='kw-summary'>
         {
            participants.map((participant, i) => {
               const nodes = i > 0 ? [<VersusIcon />]
                  : [];

               return nodes.concat([<ParticipantSummary key={participant.id}>
                  {
                     participant.lastEvents.map(event =>
                        <EventResultIndicator key={event.start} result={event.result} />
                     )
                  }
               </ParticipantSummary>]);
            })
         }
      </div>
   );
};

Summary.propTypes = {
   /**
    * The participants
    */
   participants: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
};

export default Summary;
