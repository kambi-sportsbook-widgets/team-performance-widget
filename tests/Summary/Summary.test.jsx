/* eslint-env jest */
import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';
import Summary from '../../src/js/Components/Summary';
import mockData from '../mockData.json';

let renderer;

describe('Summary DOM rendering', () => {
   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });
   it('renders correctly for mock data', () => {
      expect(renderer.render(
         <Summary participants={mockData} />
      )).toMatchSnapshot();
   });
});
