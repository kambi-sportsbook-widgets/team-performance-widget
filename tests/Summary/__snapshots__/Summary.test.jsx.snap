// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary DOM rendering renders correctly for mock data 1`] = `
<div
  className="kw-summary"
>
  <ParticipantSummary>
    <EventResultIndicator
      result="lose"
    />
    <EventResultIndicator
      result="win"
    />
    <EventResultIndicator
      result="win"
    />
    <EventResultIndicator
      result="lose"
    />
    <EventResultIndicator
      result="draw"
    />
    <EventResultIndicator
      result="draw"
    />
  </ParticipantSummary>
  <VersusIcon />
  <ParticipantSummary>
    <EventResultIndicator
      result="win"
    />
    <EventResultIndicator
      result="lose"
    />
    <EventResultIndicator
      result="lose"
    />
    <EventResultIndicator
      result="win"
    />
    <EventResultIndicator
      result="win"
    />
    <EventResultIndicator
      result="win"
    />
  </ParticipantSummary>
</div>
`;
